### ISSUE TITLE
Unexpected behavior with the `--noConfigFiles` flag

### ISSUE THREAD

**Issue by joaopalmeiro** [tclindner/npm-package-json-lint#1260]
Hi! 👋 

I was trying out the CLI and when reading the [documentation](https://npmpackagejsonlint.org/docs/cli#--noconfigfiles-alias--ncf), I interpreted that if I used the `--noConfigFiles` flag, I would be able to run the tool (with _default configuration/rules_) in a project where I don't yet have a configuration file. Looking at the [CLI implementation](https://github.com/tclindner/npm-package-json-lint/blob/v7.1.0/src/cli.ts#L54C7-L54C20), this flag appears to be _ignored_. What is the expected behavior for this flag? Is this a bug?

## Environment

- npm-package-json-lint: 7.1.0
- npm: 9.8.1
- Node: v18.18.2

### Default

```bash
npx npmPkgJsonLint .
```

#### Output

```plain
No npm-package-json-lint configuration found.
/Users/joaopalmeiro/Documents/GitHub/prettier-config/package.json
```

### `--noConfigFiles` flag

```bash
npx npmPkgJsonLint --noConfigFiles .
```

or

```bash
npx npmPkgJsonLint . --noConfigFiles
```

#### Output

```plain
No npm-package-json-lint configuration found.
/Users/joaopalmeiro/Documents/GitHub/prettier-config/package.json
```

**Comment by joaopalmeiro** [comment]
After adding a configuration file with only the author rule (and removing the author from the `package.json` file):

```json
{
    "rules": {
        "require-author": "error"
    }
}
```

```plain
npx npmPkgJsonLint .

./package.json
✖ require-author - node: author - author is required
1 error
0 warnings
```

and

```plain
npx npmPkgJsonLint --noConfigFiles .

./package.json
✖ require-author - node: author - author is required
1 error
0 warnings
```
